src/instructlab/model/backends/backends.py

```diff
@@ -167,24 +166,16 @@ def is_model_gguf(model_path: pathlib.Path) -> bool:
     # Third Party
     from gguf.constants import GGUF_MAGIC
 
-    try:
-        with open(model_path, "rb") as f:
-            # Memory-map the file on the first 4 bytes (this is where the magic number is)
-            mmapped_file = mmap.mmap(f.fileno(), length=4, access=mmap.ACCESS_READ)
-
-            # Read the first 4 bytes
-            first_four_bytes = mmapped_file.read(4)
+    if not model_path.is_file():
+        return False
 
-            # Convert the first four bytes to an integer
-            first_four_bytes_int = int(struct.unpack("<I", first_four_bytes)[0])
+    with model_path.open("rb") as f:
```

Comment:
Should we specify the encoding here?

Reply:
Binary read ("b") means it [won't](https://docs.python.org/3/library/functions.html#open) be interpreted.

```
As mentioned in the Overview, Python distinguishes between binary and text I/O. Files opened in binary mode (including 'b' in the mode argument) return contents as bytes objects without any decoding. In text mode (the default, or when 't' is included in the mode argument), the contents of the file are returned as str, the bytes having been first decoded using a platform-dependent encoding or using the specified encoding if given.
```